### PR TITLE
[ipcamera] Improve support for newer 2k+ Instar cameras

### PR DESCRIPTION
**Pull request by Skinah**
Changes made:
+ Add Person, Vehicle and Animal Object detection alarms for new Instar cameras that support this ability.
+ Add support for setting up the alarm server on new Instar range of cameras.
+ Add support for changes in Instar alarm codes in their newer range.
+ Add a method to stop ONVIF from connecting if a camera has an issue by setting onvifPort=0
+ Add compatability fix/work around for cameras that do not surround a realm in quotes.

Jar can be RIGHT HAND CLICKED and downloaded from:
https://openhab.jfrog.io/artifactory/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.ipcamera/3.4.0-SNAPSHOT/org.openhab.binding.ipcamera-3.4.0-SNAPSHOT.jar

Signed-off-by: Matthew Skinner <matt@pcmus.com>

### CODE REVIEW

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java [hunk 1]
```diff
@@ -186,26 +186,49 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     public void alarmTriggered(String alarm) {
         ipCameraHandler.logger.debug("Alarm has been triggered:{}", alarm);
-        switch (alarm) {
+        String[] queries = alarm.split("&");
+        // older cameras placed the & for the first query, whilst newer cameras do not.
+        switch (queries[0]) {
             case "/instar?&active=1":// The motion area boxes 1-4
```
Comment (lolodomo) [inline bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java:192]: As you are splitting on "&", I have the feeling these values are no more possible?

Comment (Skinah) [reply bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java:192]: Thank you for picking this up, will fix tonight.

Comment (Skinah) [reply bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java]: Fixed.

**Comment by Skinah** [comment]
All review comments have been addressed.

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java [hunk 2]
```diff
@@ -185,26 +185,50 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+                    break;
+                case "2":// car/vehicles
+                    ipCameraHandler.motionDetected(CHANNEL_CAR_ALARM);
+                    break;
+                case "3":// animals
+                case "4":
+                case "5":
+                    ipCameraHandler.motionDetected(CHANNEL_ANIMAL_ALARM);
+                    break;
+            }
```
Comment (lolodomo) [inline bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java:231]: Maybe you can add a default with a DEBUG log like you did just above for alarmCode ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java (hunk 2 above)
Comment (Skinah) [reply bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java:234]: Added unknown logging as suggested. All review comments are completed.

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java [hunk 3]
```diff
@@ -185,26 +185,53 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+                case "2":// car/vehicles
+                    ipCameraHandler.motionDetected(CHANNEL_CAR_ALARM);
+                    break;
+                case "3":// animals
+                case "4":
+                case "5":
+                    ipCameraHandler.motionDetected(CHANNEL_ANIMAL_ALARM);
+                    break;
+                default:
+                    ipCameraHandler.logger.debug("Unknown object detection code:{}", alarmCode);
```
Comment (lolodomo) [inline bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/InstarHandler.java:233]: `objectCode` and not `alarmCode`

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**